pkgs/by-name/au/audit/package.nix

```diff
@@ -25,6 +25,14 @@ stdenv.mkDerivation (finalAttrs: {
     hash = "sha256-1dG11Q7kotDReHW8aua9an1bNNlVfqhHo5+uxTH6qgo=";
   };
 
+  patches = [
+    (fetchpatch {
```

Comment:
Each patch in nixpkgs applied to the upstream source should be documented. Out-of-tree patches, fetched using `fetchpatch`, are preferred.

In each patch comment, please explain the purpose of the patch and link to the relevant upstream issue if possible. If the patch has been merged upstream but is not yet part of the released version, please note the version number or date in the comment such that a future maintainer updating the nix expression will know whether the patch has been incorporated upstream and can thus be removed from nixpkgs.

Furthermore, please use a _stable_ URL for the patch. Rather than, for example, linking to a GitHub pull request of the form `https://github.com/owner/repo/pull/pr_number.patch`, which would change every time a commit is added or the PR is force-pushed, link to a specific commit patch in the form `https://github.com/owner/repo/commit/sha.patch`.

Here are two good examples of patch comments:

```nix
mkDerivation {
  …

  patches = [
    # Ensure RStudio compiles against R 4.0.0.
    # Should be removed next 1.2.X RStudio update or possibly 1.3.X.
    (fetchpatch {
      url = "https://github.com/rstudio/rstudio/commit/3fb2397c2f208bb8ace0bbaf269481ccb96b5b20.patch";
      sha256 = "0qpgjy6aash0fc0xbns42cwpj3nsw49nkbzwyq8az01xwg81g0f3";
    })
  ];
}
```

```nix
mkDerivation {
  …

  patches = [
    # Allow building with bison 3.7
    # PR at https://github.com/GoldenCheetah/GoldenCheetah/pull/3590
    (fetchpatch {
      url = "https://github.com/GoldenCheetah/GoldenCheetah/commit/e1f42f8b3340eb4695ad73be764332e75b7bce90.patch";
      sha256 = "1h0y9vfji5jngqcpzxna5nnawxs77i1lrj44w8a72j0ah0sznivb";
    })
  ];
}
```

Reply:
I think it would be more polite to just [link the document you’re copying from](https://github.com/jtojnar/nixpkgs-hammering/blob/922ec48e954235772735191acc9f2cefeb7b6604/explanations/missing-patch-comment.md) rather than pasting the whole thing on a PR by a committer of many years’ experience, especially when the majority of it is irrelevant in this case. (BTW, the advice about being able to get stable commit URLs for PRs is incorrect, as Alyssa has pointed out on numerous PRs in the past :) )

(Personally I think a good patch name, like a good variable name, is adequate documentation when the comment would simply restate the name. Admittedly `static.patch` is not the most verbose, but I think there is no ambiguity about what a patch named `static.patch` is there for. Patches with an opaque URL, no comment, and no filename are a much bigger issue though, of course.)

Reply:
I'm having default replies enabled in GitHub so that I can do reviews faster, this is why. I'm pretty well aware that you're a long time contributor, and you should not feel offended by this message. I was on the phone (just like now) and it was easier for me to use a default reply. Also, I approved the PR without requiring change, meaning that it was not a blocking point. Hope no hard feelings?

Reply:
This is Alyssa’s PR, not mine, FWIW :)

I agree a comment would be nice, though hopefully upstream acts quickly on this so it doesn’t last too long anyway.

Reply:
Most of this seems pretty irrelevant to my PR?  Having a template is okay, but it would be nice to at least cut out the irrelevant bits when you use it.

I think it's better to set the name of the patch than have a comment, because the name also shows up when building, and reviews in the past have suggested changing things in this direction (comment to name).

Reply:
IMO, it's difficult for a comment about adding a patch to say anything useful — I can't necessarily say "remove on next release", because maybe they do a release that branches off the current one and doesn't include this patch.  So all I can really say with near certainty is that this commit will be in some future release, which I'm not sure is really helpful.  WDYT?

Reply:
Point taken, you're right. To avoid the issue in the future, I'll refrain myself from reviewing from my phone and wait to be fully woken up before doing a review :)
Feel free to commit this PR, it was not a blocking point at all for me.